google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/SpannerClientHeaderProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spanner.spi.v1;
+
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.common.collect.ImmutableMap;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class SpannerClientHeaderProvider implements HeaderProvider {
```

Comment:
this implements `HeaderProvider` but I don't think `getHeaders` from this class is being called anywhere.
It might be better to just put constructHeadersAsMetadata and getResourceHeadersAsMetadata directly in GrpcSpannerRpc class

Reply:
It is easier to test it if it is a separate class. Also I would like to keep headers stuff isolated, this has a good chance to be useful during Spanner GA work, when (if?) it gets migrated to gapic stubs.

Reply:
Ok, it can be a separate class but there is no need to implemented HeaderProvider for now. That can be added later if needed.